packages/beeq/src/components/progress/bq-progress.tsx

```diff
@@ -0,0 +1,184 @@
+  // =======================
+  @Watch('mode')
+  @Watch('thickness')
+  @Watch('type')
+  handleTypePropChange() {
+    validatePropValue(PROGRESS_MODE, 'determinated', this.el, 'mode');
+    validatePropValue(PROGRESS_THICKNESS, 'medium', this.el, 'thickness');
+    validatePropValue(PROGRESS_TYPE, 'default', this.el, 'type');
+  }
+
+  @Watch('value')
+  handleValuePropChange(newValue: number) {
+    this.validateValue(newValue);
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.handleTypePropChange();
+    this.handleValuePropChange(this.value);
+  }
+
+  componentDidUpdate() {
+    this.checkIsIndeterminated();
+    this.setProgressIndeterminate();
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private setProgressIndeterminate = () => {
+    const hasValueAttribute = this.el.hasAttribute('value');
+    const hasPreviousValue = this.previousValue !== null;
+
+    if (this.mode === 'indeterminate' && hasValueAttribute) {
+      this.previousValue = this.value;
+      this.el.removeAttribute('value');
+      return;
+    }
+
+    if (this.mode !== 'indeterminate' && hasPreviousValue) {
+      this.el.setAttribute('value', `${this.previousValue}`);
+    }
+  };
+
+  private checkIsIndeterminated() {
+    const isIndeterminated = this.mode === 'indeterminate';
+    if (isIndeterminated) {
+      this.enableTooltip = false;
+      this.label = false;
+    }
+    return isIndeterminated;
+  }
+
+  private validateValue(newValue: number) {
+    if (this.value) {
+      const clampedValue = Math.max(0, Math.min(100, newValue));
+      if (newValue !== clampedValue) {
+        this.value = clampedValue;
+      }
+    }
+    return this.value;
+  }
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    const progressClasses = {
+      [`progress-bar progress-bar__${this.type} ${this.thickness}`]: true,
+      'progress-bar__border-shape rounded-full': !this.borderShape,
+      'h-1': this.thickness === 'medium',
+      'h-2': this.thickness === 'large',
+      indeterminate: this.checkIsIndeterminated(),
+    };
+
+    return (
+      <div class="flex items-center gap-xs">
+        <div class="relative flex items-center">
+          <progress class={progressClasses} value={this.value} max="100" slot="triger"></progress>
```

Comment:
Is there any reason why we should set a slot name here?

```suggestion
          <progress class={progressClasses} value={this.value} max="100"></progress>
```

Reply:
Nope, I'll remove it. Added it just by mistake.